src/com/google/javascript/jscomp/AbstractCommandLineRunner.java

```diff
@@ -579,12 +604,47 @@ public FlagUsageException(String message) {
    *
    * @param files A list of filenames.
    * @param zips A list of zip filenames.
+   * @param jsonFiles A list of json encoded files.
+   * @return An array of inputs
+   */
+  protected List<SourceFile> createInputs(List<String> files,
+      List<String> zips, List<JsonFileSpec> jsonFiles) throws FlagUsageException, IOException {
```

Comment:
Do you need to combine json inputs with zips or files? If not, you can have a createInputs just for the jsons, and it populates inputs directly, it doesn't call into another createInputs method.